### PR TITLE
Impl. Service for AddTopic handling and axios call

### PR DESCRIPTION
**Pull request by timlenny**
Service that handles adding new topics and made adjustments in the frontend

### CODE REVIEW

**Review by lostspezi — CHANGES_REQUESTED** [review]
You have a Codesmell in the backend and you only tested the [TopicConversionService](https://sonarcloud.io/component_measures?id=timlenny_cgn-java-23-1-capstone_backend&pullRequest=13&metric=new_coverage&view=list&selected=timlenny_cgn-java-23-1-capstone_backend%3Asrc%2Fmain%2Fjava%2Fcom%2Ftimlenny%2Fbackend%2Fservice%2FTopicConversionService.java) particually. For a review contact us in slack if you have fixed it.

**Review by HendrikNeuschulz — APPROVED** [review]
for me, there is nothing to change here.

File: backend/src/main/java/com/timlenny/backend/repository/TopicRepository.java [hunk 1]
```diff
@@ -4,6 +4,9 @@
 import org.springframework.data.mongodb.repository.MongoRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.Optional;
+
 @Repository
 public interface TopicRepository extends MongoRepository<Topic, String> {
+    Optional<Topic> findByTopicName(String topicName);
 }
```
Comment (lostspezi) [inline backend/src/main/java/com/timlenny/backend/repository/TopicRepository.java:12]: no new line

**Review by lostspezi — APPROVED** [review]
lgtm

**Pull request merged**